### ISSUE TITLE
`Lint/LiteralAsCondition` makes invalid autocorrections

### ISSUE THREAD

**Issue by nevans** [rubocop/rubocop#13916]
After upgrading to 1.73.0, Lint/LiteralAsCondition is causing the following autocorrections:

```ruby
- h.label_tag(name, class: errors[name.to_sym] && "error")
+ h.label_tag(name, class: errors[name.to_sym])
```

```ruby
 def message
   "Remote provider %s returned %s%s" % [
-    provider, result, desc && " (#{desc})",
+    provider, result, desc,
   ]
 end
```

In both cases, the value of the expression is used for more than just a boolean check. This would be safe for boolean expressions in conditionals (`if bool`, `elsif bool`, `unless bool`, `while bool`, `until bool`, `bool ? expr1 : expr2`).  But for anything else we may care about the distinction between `nil` vs `false` or `true` vs other truthy values.

Could this be related to #13117?

--------

## Expected behavior

These should not be reported as offenses, nor should they be autocorrected.

## Actual behavior

They are reported as offenses and incompatibly autocorrected.

## Steps to reproduce the problem

See examples above.

## RuboCop version

1.73.0

**Comment by rmm5t** [comment]
Just came here to report the same unsafe autocorrections. 

At a minimum, this cop's autocorrections cannot be marked as "safe."

**Comment by Zopolis4** [comment]
Ah. Yep, I can see how that's an issue. I'll look into a fix.
